x-pack/plugins/security/server/session_management/session_index.ts

```diff
@@ -403,10 +403,15 @@ export class SessionIndex {
             }
           );
         } catch (err) {
-          this.options.logger.error(
-            `Failed to check if session legacy index template exists: ${err.message}`
-          );
-          return reject(err);
+          // The Template API is deprecated and may become unavailable at some point (404 Not Found). It's also
+          // unavailable in the Serverless offering (410 Gone). In either of these cases, we should disregard the error.
+          const errorStatusCode = getErrorStatusCode(err);
+          if (errorStatusCode !== 404 && errorStatusCode !== 410) {
```

Comment:
nit: I see a test added for the 410 condition, but it would be great to have one for the 404 condition as well!

Reply:
@azasypkin I see the test is now added - LGTM!